apis/v1alpha1/validation/validation_test.go

```diff
@@ -160,3 +160,275 @@ func TestValidateGateway(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateHTTPRoute(t *testing.T) {
+	testService := "test-service"
+	specialService := "special-service"
+	tests := []struct {
+		name     string
+		hRoute   gatewayv1a1.HTTPRoute
+		errCount int
+	}{
+		{
+			name: "valid httpRoute with no filters",
+			hRoute: gatewayv1a1.HTTPRoute{
+				Spec: gatewayv1a1.HTTPRouteSpec{
+					Rules: []gatewayv1a1.HTTPRouteRule{
+						{
+							Matches: []gatewayv1a1.HTTPRouteMatch{
+								{
+									Path: &gatewayv1a1.HTTPPathMatch{
+										Type:  pathMatchTypePtr("Prefix"),
+										Value: stringPtr("/"),
+									},
+								},
+							},
+							ForwardTo: []gatewayv1a1.HTTPRouteForwardTo{
+								{
+									ServiceName: &testService,
+									Port:        portNumberPtr(8080),
+									Weight:      int32Ptr(100),
+								},
+							},
+						},
+					},
+				},
+			},
+			errCount: 0,
+		},
+		{
+			name: "valid httpRoute with 1 filter",
+			hRoute: gatewayv1a1.HTTPRoute{
+				Spec: gatewayv1a1.HTTPRouteSpec{
+					Rules: []gatewayv1a1.HTTPRouteRule{
+						{
+							Matches: []gatewayv1a1.HTTPRouteMatch{
+								{
+									Path: &gatewayv1a1.HTTPPathMatch{
+										Type:  pathMatchTypePtr("Prefix"),
+										Value: stringPtr("/"),
+									},
+								},
+							},
+							Filters: []gatewayv1a1.HTTPRouteFilter{
+								{
+									Type: "RequestMirror",
```

Comment:
nit: Maybe use the const type here and for all the other filter types?: `HTTPRouteFilterRequestMirror`